File: docs/code-examples/depth_image_3d.rs
//! Create and log a depth image.
use ndarray::{s, Array, ShapeBuilder};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) =
        rerun::RecordingStreamBuilder::new("rerun_example_depth_image").memory()?;

    let mut image = Array::<u16, _>::from_elem((200, 300).f(), 65535);
    image.slice_mut(s![50..150, 50..150]).fill(20000);
    image.slice_mut(s![130..180, 100..280]).fill(45000);

    let depth_image = rerun::DepthImage::try_from(image.clone())?.with_meter(10000.0);

    // If we log a pinhole camera model, the depth gets automatically back-projected to 3D
    rec.log(
        "world/camera",
        &rerun::Pinhole::from_focal_length_and_resolution(
            [20.0, 20.0],
            [image.shape()[1] as f32, image.shape()[0] as f32],
        ),
    )?;

    rec.log("world/camera/depth", &depth_image)?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}

File: docs/code_examples/depth_image_3d.py
"""Create and log a depth image and pinhole camera."""

import numpy as np
import rerun as rr

# Create a dummy depth image
image = 65535 * np.ones((8, 12), dtype=np.uint16)
image[0:4, 0:6] = 20000
image[4:8, 6:12] = 45000


rr.init("rerun_example_depth_image", spawn=True)

# If we log a pinhole camera model, the depth gets automatically back-projected to 3D
rr.log(
    "world/camera",
    rr.Pinhole(
        width=image.shape[1],
        height=image.shape[0],
        focal_length=20,
    ),
)

# Log the tensor.
rr.log("world/camera/depth", rr.DepthImage(image, meter=10_000.0))
